Adding animal orgs

NunoSempere opened the pull request.

NunoSempere committed:
tweak: rework plans

NunoSempere committed:
feat: add another animal org: Open Wing Alliance. And rework plans.

OAGr commented on attempt_6_animals/simple-reference-animals.squiggle:
```diff
@@ -138,4 +118,44 @@ value_fwi_shrimp =
+
+What am I estimating: Impact of corporate commitments that OWA made happen in 2022.
+
+Notes on the estimation: The number of corporate campaigns is from OWA's own reporting. The number of chickens affected and the improvement per chicken is pretty much pulled out of my ass.
```
> The number of chickens affected and the improvement per chicken is pretty much **pulled out of my ass.**

Can we have some other terminology here? Maybe use "levels of how speculative something is" or similar?

Lots of stuff here will be very highly speculative.

NunoSempere replied:
Yes

NunoSempere committed:
fix: vocab.

NunoSempere committed:
feat: savepoint, add more animal orgs

NunoSempere committed:
tweak: wrangle estimates into relative value shape

NunoSempere committed:
tweak: one more step towards shaping as relative values

NunoSempere committed:
tweak: use all animal estimates.

OAGr commented on attempt_6_animals/simple-reference-animals.squiggle:
```diff
@@ -29,16 +10,17 @@ value_farmed_cow_year_in_human_qalys = normal({ p10: -0.2, p90: 0.1 })
 weight_cow = mixture([450 to 1800, 360 to 1100], [1/2,1/2])
+wastage = 0.3 to 0.5 // should really be a beta
 lifetime_cow = (30 to 42) / 12
 calories_cow = mixture(0.8M to 1.4M, (500k to 700k) * weight_cow/1000) 
```
Note: Any dist that gets used by more that one later dist, should probably be made into a SampleSet. That will give you more correlations.

NunoSempere replied:
Yes. It seems like I missed wastage in this case, though.

OAGr commented on attempt_6_animals/simple-reference-animals.squiggle:
```diff
@@ -126,16 +111,260 @@ improvement_as_proportion_of_lifetime = 0.05 to 0.5
   fish_potentially_helped * 
 	improvement_as_proportion_of_lifetime * 
-	(value_happy_salmon_year_in_human_qalys / salmon_estimates.lifetime)
+	(salmon_estimates.value_year_in_human_qalys / salmon_estimates.lifetime)
```
I suggest valuing salmon in some fish/animal unit, for the correlations to work.

OAGr commented on attempt_6_animals/simple-reference-animals.squiggle:
```diff
@@ -126,16 +111,260 @@ improvement_as_proportion_of_lifetime = 0.05 to 0.5
+}
+
+/*
+## Open Wing Alliance
```
A bit later on, I could see having the Animal evaluators in a separate file/component, but defining them in terms of a few passed-in variables for the animals.

NunoSempere replied:
Agree, but probably not yet.

OAGr commented on attempt_6_animals/simple-reference-animals.squiggle:
```diff
@@ -126,16 +111,260 @@ improvement_as_proportion_of_lifetime = 0.05 to 0.5
+average_duration_fulfilled_commitments = 1 to 10 // years. Could be more?
+average_reduction_in_suffering_of_commitment_per_chicken = 0.05 to 0.3 // pulled out of my ass.
+
+value_owa_2022 = num_corporate_commitments * 
```
Here you're using one inside-view model, to represent the entire org. I'd also recommend mixing this with maybe 5% mass, on a very wide intuitive estimate, or similar. Otherwise you could really be hurt when scored. (You need to watch the tails)

NunoSempere replied:
Good point, done.

OAGr commented on attempt_6_animals/simple-reference-animals.squiggle:
```diff
@@ -126,16 +111,260 @@ improvement_as_proportion_of_lifetime = 0.05 to 0.5
+
+num_corporate_commitments = 150 to 200 
+// ^ "nearly 200 commitments", from https://thehumaneleague.org/article/owa-year-in-review-2022
+average_num_chickens_per_year_per_commitment = 5k to 20k 
```
I suggest using the word "mean" instead of "average", to be clearer.

NunoSempere replied:
Done

OAGr commented on attempt_6_animals/simple-reference-animals.squiggle:
```diff
@@ -126,16 +111,260 @@ improvement_as_proportion_of_lifetime = 0.05 to 0.5
+// mentions 40k, but also seems to be the most impressive one for 2022 (?)
+frequency_promises_are_fulfilled = 0.4 to 0.9
+average_duration_fulfilled_commitments = 1 to 10 // years. Could be more?
+average_reduction_in_suffering_of_commitment_per_chicken = 0.05 to 0.3 // pulled out of my ass.
```
I don't like this phrase in code, we'll be using it a lot, and this code will be very visible.

NunoSempere replied:
Changed to "pulled out of thin air"

OAGr commented on attempt_6_animals/simple-reference-animals.squiggle:
```diff
@@ -126,16 +111,260 @@ improvement_as_proportion_of_lifetime = 0.05 to 0.5
+  result = [
+    ({name: animal.name + " value per year of life", value: value_year_in_human_qalys}),
+    ({name: animal.name + " value per lifetime", value: value_in_lifetime_human_qalys}),
+    ({name: animal.name + " value per calorie", value: value_per_calorie_human_qalys}),    
```
IDs follow certain naming conventions, they don't have spaces and stuff. You can see all the IDs [here](https://relative-values-git-develop-quantified-uncertainty.vercel.app/interfaces/macrovariables/about).

OAGr replied:
Don't worry about having nice names, in this section. You just need to match IDs. Names are separately attached to IDs elsewhere.

NunoSempere replied:
Ok, adding some really very lazy ids for now. Checking, I think that the name is used for most things though.

NunoSempere replied:
Added some better ids now.

NunoSempere committed:
tweak: wastage of cow now reused, rename, and fed to ss.

NunoSempere committed:
tweak: add uncertainty

should probably refactor orgs into their own function, but
I'm thinking not yet.

NunoSempere committed:
tweak: add ids, start to try to add uncertainty

NunoSempere committed:
tweak: figure out how to add uncertainty, tweak ids a bit.

NunoSempere committed:
tweak: add script to detect repeated vars, ss a few repeated vars.

NunoSempere committed:
tweak: threshold of 0 for object vars

I want them to also be correlated with the original object!

OAGr commented on attempt_6_animals/simple-reference-animals.squiggle:
```diff
@@ -363,7 +373,18 @@ organization_items = [
+  helper_list = List.upTo(0, l-1)
+  result = List.map(helper_list, {|i| 
+    obj = xs[i]
+    obj2 = ({id: "relative_value_animals_2023_04_n" + i})
```
You're going to need to still map these to legit names and stuff. My guess is that this method will be a pain for that.

Later on, the IDs will be the key part of the forecast, so important to not have change. (Or, later, if we allow for a way to roll back an estimate)

NunoSempere replied:
These are already mapped to legit names, though.

OAGr replied:
Do you mean the variable names? Or that they have names with strings? Those wouldn't work as the IDs.

OAGr replied:
We need something to match a page like this (the data still needs to be put together, perhaps by this model)

<img width="1511" alt="image" src="https://user-images.githubusercontent.com/377065/233738259-f1a200bc-d92e-4c71-817a-c22e9b51b39b.png">

Look at how this is done. We need something similar.
https://github.com/quantified-uncertainty/squiggle/blob/develop/packages/relative-values/models/src/macrovariables.ts

OAGr replied:
I saw how this works in /squiggle. If you match them, it does work for now, so you don't need to worry about this now. But later on, we could definitely want a change here, if we use this in larger environments.

NunoSempere replied:
Ok, invested a few mins into having proper ids.

NunoSempere committed:
tweak: start adding real ids to animals

NunoSempere committed:
tweak: experiment with using sed for adding data

NunoSempere committed:
Revert "tweak: experiment with using sed for adding data"

This reverts commit 300116acd47dfa4e1ad3f7a8ad690f8dd37bdf99.

NunoSempere committed:
tweak: get more sensible ids.

NunoSempere committed:
tweak: fix error after changing the ids.
